### PR TITLE
[FLINK-26955] Add CRD compatibility check to maven build

### PR DESCRIPTION
**Pull request by gyfora**
Add automated CRD compatibility check logic to the maven build comparing against the 1.0.0 release.

Manually verified + unit tests added for the cases covered by the checker

### CODE REVIEW

**Comment by gyfora** [comment]
cc @bgeng777 @morhidi

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/CrdCompatibilityChecker.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Utility for checking backward CRD compatibility. */
```
Comment (tweise) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/CrdCompatibilityChecker.java:32]: Maybe add a summary here on how compatibility is checked?

**Comment by morhidi** [comment]
Thanks @gyfora for the PR. Having such a tool overall is a great idea, it'll force us to think twice before introducing any changes to the CRD. I second @tweise's suggestion to agree formally on a few bullets that we add to the CRD checker, e.g.: https://www.faun.dev/c/stories/dineshparvathaneni/kubernetes-crd-versioning-for-operator-developers/

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/CrdCompatibilityChecker.java (hunk 1 above)
Comment (gyfora) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/CrdCompatibilityChecker.java]: Done

**Comment by morhidi** [comment]
+1 LGTM, as we understand more about CRD version compatibility we should add/modify the rules here. Nice job!

**Pull request merged**